packages/toolkit/src/codewhisperer/service/transformByQHandler.ts

```diff
@@ -78,89 +77,123 @@ export async function getOpenProjects() {
-        telemetry.codeTransform_isDoubleClickedToTriggerInvalidProject.emit({
-            codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
-            codeTransformPreValidationError: 'NonMavenProject',
-            result: MetadataResult.Fail,
-            reason: buildSystem === BuildSystem.Gradle ? buildSystem : 'NoPomFileFound',
-        })
-        throw new ToolkitError('No valid Maven build file found', { code: 'CouldNotFindPomXml' })
+export async function validateOpenProjects(projects: vscode.QuickPickItem[]) {
+    const javaProjects = []
+    for (const project of projects) {
+        const projectPath = project.description
+        const javaFiles = await vscode.workspace.findFiles(
+            new vscode.RelativePattern(projectPath!, '**/*.java'),
+            '**/node_modules/**',
+            1
+        )
+        if (javaFiles.length > 0) {
+            javaProjects.push(project)
+        }
     }
-    const compiledJavaFiles = await vscode.workspace.findFiles(
-        new vscode.RelativePattern(projectPath!, '**/*.class'),
-        '**/node_modules/**',
-        1
-    )
-    if (compiledJavaFiles.length < 1) {
+    if (javaProjects.length === 0) {
         void vscode.window.showErrorMessage(CodeWhispererConstants.noSupportedJavaProjectsFoundMessage)
         telemetry.codeTransform_isDoubleClickedToTriggerInvalidProject.emit({
             codeTransformSessionId: codeTransformTelemetryState.getSessionId(),
             codeTransformPreValidationError: 'NoJavaProject',
             result: MetadataResult.Fail,
-            reason: 'NoJavaProjectsAvailable',
+            reason: 'CouldNotFindJavaProject',
         })
-        throw new TransformByQJavaProjectNotFound()
-    }
-    const classFilePath = `${compiledJavaFiles[0].fsPath}`
-    const baseCommand = 'javap'
-    const args = ['-v', classFilePath]
-    const spawnResult = spawnSync(baseCommand, args, { shell: false, encoding: 'utf-8' })
-    if (spawnResult.status !== 0) {
-        void vscode.window.showErrorMessage(CodeWhispererConstants.noSupportedJavaProjectsFoundMessage)
-        let errorLog = ''
-        errorLog += spawnResult.error ? `${JSON.stringify(spawnResult.error)}` : ''
-        errorLog += `${spawnResult.stderr}\n${spawnResult.stdout}`
-        getLogger().error(`CodeTransform: Error in running javap command = ${errorLog}`)
-        let errorReason = ''
-        if (spawnResult.stdout) {
-            errorReason = 'JavapExecutionError'
-            // should never happen -- stdout from javap has always been much, much smaller than the default buffer limit of 1MB
-            if (Buffer.byteLength(spawnResult.stdout, 'utf-8') > CodeWhispererConstants.maxBufferSize) {
-                errorReason += '-BufferOverflow'
-            }
-        } else {
-            errorReason = 'JavapSpawnError'
+        throw new ToolkitError('No Java projects found', { code: 'CouldNotFindJavaProject' })
+    }
+    const mavenJavaProjects = []
```

Comment:
Possibly the same here to make more modular code. Something like `const mavenJavaProjects = getMavenJavaProjects()`

Reply:
Fixed